src/components/Banner/alert_templates.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+import { makeGitHubIssueLink } from "../../colorizer/utils/url_utils";
+import { VERSION_NUMBER } from "../../constants";
+
+import { AlertBannerProps } from "./AlertBanner";
+
+const MAX_ERROR_MESSAGE_LENGTH = 50;
+
+/** Trims whitespace at the start and end of each line in a multiline string. */
+const trimMultilineString = (str: string): string => {
+  return str
+    .split("\n")
+    .map((line) => line.trim())
+    .join("\n");
+};
+
+/**
+ * Shows an alert banner for when viewer settings could not be loaded
+ * from the URL. Includes a link to report the issue on GitHub.
+ */
+export const showFailedUrlParseAlert = (url: string, error: Error): AlertBannerProps => {
+  let errorMessage = error.message;
+  if (errorMessage.length > MAX_ERROR_MESSAGE_LENGTH) {
+    errorMessage = errorMessage.slice(0, MAX_ERROR_MESSAGE_LENGTH) + "...";
+  }
+  // Break URL into lines for each param for better readability
+  url = url.replace("?", "?\n");
+  url = url.replaceAll("&", "\n&");
+
+  const githubIssueLink = makeGitHubIssueLink(
+    `Viewer settings could not be loaded from the URL: \`${errorMessage}\``,
+    trimMultilineString(`Initial loading of viewer settings from the URL failed unexpectedly.
+
+    **Please provide any additional information that may be helpful in diagnosing the issue:**
+      
+
+
+    ---
+    *[The following contents were auto-generated by TFE.]*
+
+
+    **URL:**
+    \`\`\`
+    ${url}
+    \`\`\`
+
+    **Error:**
+    \`\`\`
+    ${error}
+    \`\`\`
+
+    **Stack trace:**
+    \`\`\`
+    ${error.stack}
```

Comment:
is `error.stack` always defined?

Reply:
Hmm, good question. It looks like MDN has conflicting notes about it. It's not strictly standardized but is usually defined in most JS environments. I think it's fine to include, since if it's not defined it'll just return `undefined` as a string.